app/src/main/java/com/example/voyagerx/util/SharedPreferencesManager.kt

```diff
@@ -0,0 +1,81 @@
+package com.example.voyagerx.util
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+
+class SharedPreferencesManager(val context: Context) {
+    private val sharedPref: SharedPreferences = context.getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)
+    private val editor = sharedPref.edit()
+
+    private fun String.put(long: Long) {
+        editor.putLong(this, long)
+        editor.apply()
+    }
+
+    private fun String.put(int: Int) {
+        editor.putInt(this, int)
+        editor.apply()
+    }
+
+    private fun String.put(string: String) {
+        editor.putString(this, string)
+        editor.apply()
+    }
+
+    private fun String.put(boolean: Boolean) {
+        editor.putBoolean(this, boolean)
+        editor.apply()
+    }
+
+    private fun String.getLong() = sharedPref.getLong(this, 0)
+    private fun String.getInt() = sharedPref.getInt(this, 0)
+    private fun String.getString() = sharedPref.getString(this, "")!!
+    private fun String.getBoolean() = sharedPref.getBoolean(this, true)
+
+    fun setBackgroundWallpaper(wallpaperSelection: Boolean) = PREF_BKG_APPEARANCE.put(wallpaperSelection)
+
+    fun getBackgroundWallpaper() : Boolean {
+        PREF_BKG_APPEARANCE.getBoolean()
+        Log.d("appearance switch", PREF_BKG_APPEARANCE.getBoolean().toString())
+        return PREF_BKG_APPEARANCE.getBoolean()
+    }
+
+    fun setTextInDropdown(dropDownText: String) = PREF_DROP_DOWN_TEXT.put(dropDownText)
+
+    fun getTextInDropdown() : String {
+        PREF_DROP_DOWN_TEXT.getString()
```

Comment:
This raises an interesting question to me being new to Kotlin that I'm not 100% sure on. Is it generally an acceptable practice to add methods to these primitive objects like String, Boolean, etc? When I first read this, I kind of interpreted it as every String in the global scope would now have this getString() method available. But now I see it's a private method so I'm guessing this is just for this class? It looks like you're re-using the logic of fetching preferences based on primitive type which I like.

Reply:
Hey, @chrsbell. You know, I am not really sure if that's considered best practice or not. The SharedPreferences example had the variables being put as their data type, so this Manager could probably be simplified and skip a step. You are correct that since the String.get calls are private, the only functions that can be referenced outside of the SharedPreferences manager are the specific get/set for wallpaper, font size, etc.